### ISSUE TITLE
当option项为空值时，会出现parse err错误，也无法解析到想要的对象

### ISSUE THREAD

**Issue by jalin107** [digineo/go-uci#13]

**Comment by dmke** [comment]
I'm sorry, I don't speak nor read chinese. Google translates your issue into:

> When the option item is empty, a parse err error will occur and the desired object cannot be parsed

Is that translation correct? If so, what do you mean by "when the option item is emtpy"?

**Comment by jalin107** [comment]
yes.

**Comment by dmke** [comment]
I have identified this behaviour on my OpenWRT router:

```console
# cat /etc/config/test
config section
	option empty1
	option empty2 ''
	list empty3
	list empty4 ''

# uci export test
package test

config section
	list empty3 ''
	list empty4 ''
```

It was easy to extend the lexer to recognize the empty/missing values properly (I personally would consider an option without value invalid.)

To fix the parser, I'll need some more time (but I don't know when that will be).  Feel free to extend on e84480d if you don't want to wait :-)

**Comment by SmartBoy84** [comment]
3 years later; any update?

**Comment by dmke** [comment]
Nope, still very low on my priorities.

Feel free to submit a patch though, I'll happily merge it.

**Comment by jjm2473** [comment]
Any progress? This library cannot parse aria2 default config.

**Comment by dmke** [comment]
No, sorry. At the moment (and for the foreseeable future), we have close to zero maintenance capacity for this. Our use-cases are covered, so active development is de-facto halted.
